include/measurement_kit/http/http.hpp

```diff
@@ -328,7 +327,7 @@ class Stream {
      */
     void on_connect(std::function<void(void)>&& fn) {
         connect_handler = fn;
-        connection->on_connect([this]() {
+        connection.on_connect([this]() {
```

Comment:
Just to understand the rational. You think the memory usage reduction is not worth complicating the code? If so, what is the benefit of not using the shared pointer?